gson/src/main/java/com/google/gson/FieldNamingStrategy.java

```diff
@@ -37,4 +40,16 @@ public interface FieldNamingStrategy {
    * @since 1.3
    */
   public String translateName(Field f);
+
+  /**
+   * Translates the field name into its JSON field alternative names representation. Used for
+   * deserialization only. This is similar to {@link SerializedName#alternate()}.
+   *
+   * @param f the field object that we are translating
+   * @return the list of possible translated field names.
+   * @since $next-version$
+   */
+  default List<String> translateToAlternateNames(Field f) {
```

Comment:
I think the existing name `translateName` is maybe not ideal. It's not really about translating the name, it's about deriving a name for the `Field`, which is _probably_ based in some way on the name of the field but could for example come from an annotation on the field. So perhaps just `alternateNames` would be a better name here.

Reply:
I have changed the method name to alternateNames